### DISCUSSION TITLE
Having More Than One Application Shell

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#11586] [General] [answered]
Hi,
Can we have more than one kind of application shell in theia that user can switch between ?

For eg. 
Application Shell 1: Normal Mode -> Vannila Theia
Application Shell 2: Debug Mode -> Different sets of custom editors, tree widgets, menu & statusbar (Entirely new sets of widgets)

Inorder to Switch maybe we could have a command registered. A service to share info between the two layouts.

I tried to follow this approach, disposing the not-needed widgets while switching to the new custom layout and re-opening them once switched back to the usual layout . However, while switching back costly init API calls are being made which deterred me from following this approach. 

When we close and reopen the theia instance in browser, Application State is saved and retrieved. Similar to this, can we store and retrieve the entire application shell while switching between different layouts ?

Kindly provide some suggestions regarding this scenario and any another different approach I could take. 

Thanks

**Comment by vince-fugnitto** [comment, marked as answer]
Closing in favor of https://github.com/eclipse-theia/theia/discussions/11744.
